### ISSUE TITLE
[Feature Request] VirusTotal integration

### ISSUE THREAD

**Issue by Spidy123222** [SideStore/SideStore#74]
The Idea is that during install or as a setting in settings. You can enable the ability for virus scanning via VirusTotal during a initial install or a update. 

VirusTotal api: https://developers.virustotal.com/reference/overview

**Comment by JJTech0130** [comment]
Is VirusTotal able to inspect IPAs? Seems like it wouldn’t have a large signature dataset to draw upon.

**Comment by Spidy123222** [comment]
> Is VirusTotal able to inspect IPAs? Seems like it wouldn’t have a large signature dataset to draw upon.

Yeah it is able to inspect ipa's. How I tested it was get a jailbreak ipa and see if it trips as a Trojan and get a normal app like dolphiniOS see if it trips. It worked fine.

**Comment by SARankDirector-Minecraft** [comment]
That might actually be really neat

**Comment by WesleyBryie** [comment]
This sounds like a great idea, one thing to consider is some upload speeds for some people are low and it would be added time to install.

**Comment by preland** [comment]
Due to the size of apps, it would likely be a good idea to have this be optional. I would have 3 separate options: No security, which means nothing gets scanned ever; medium security, which means it will send the sha256 hash to VirusTotal to get the scan result, and if VirusTotal doesn’t recognize the hash, the user is warned that the app is “unknown”, after which the user can either ignore the alert and install, send the full ipa to VirusTotal, or cancel installation; and high security, which will send the sha256 hash to VirusTotal, and if it isn’t present it will automatically send the ipa to VirusTotal.

The main limitation with using VirusTotal is that any app with a higher size than 32MB would need a dedicated API request, and any app larger than 650MB would be unable to be scanned. This could be potentially fixed by compressing the app before sending it, but this would be inconsistent in effectiveness, and there would still be a size limitation.

**Comment by Spidy123222** [comment]
Currently we might not need to use virustotal if we continue work with Debianarch with the virus tool he made which is local.
